spec.html

```diff
@@ -22,4 +22,25 @@ <h1>String.prototype.isWellFormed ( )</h1>
       1. Return IsStringWellFormedUnicode(_S_).
     </emu-alg>
   </emu-clause>
+
+  <emu-clause number="11" id="sec-string.prototype.towellformed">
+    <h1>String.prototype.toWellFormed ( )</h1>
+    <p>This method returns a String representation of this object with all leading surrogates and trailing surrogates that are not part of a surrogate pair replaced with U+FFFD (REPLACEMENT CHARACTER).</p>
+    <p>It performs the following steps when called:</p>
+    <emu-alg>
+      1. Let _O_ be ? RequireObjectCoercible(*this* value).
+      1. Let _S_ be ? ToString(_O_).
+      1. Let _strLen_ be the length of _S_.
+      1. Let _k_ be 0.
+      1. Let _result_ be the empty String.
+      1. Repeat, while _k_ &lt; _strLen_,
+        1. Let _cp_ be CodePointAt(_S_, _k_).
+        1. If _cp_.[[IsUnpairedSurrogate]] is *true*, then
+          1. Append 0xFFFD (REPLACEMENT CHARACTER) to _result_.
+        1. Else,
+          1. Set _result_ to the string-concatenation of _result_ and UTF16EncodeCodePoint(_cp_.[[CodePoint]]).
```

Comment:
```suggestion
          1. Append UTF16EncodeCodePoint(_cp_.[[CodePoint]]) to _result_.
```
or alternatively, change line 39 so it sets to the string-concatenation.

Reply:
Can't. `UTF16EncodeCodePoint` returns a string. Also can't go the other way unless we wanted to construct a string with a single code unit just to string-concat it, which seems unnecessarily obtuse.

Reply:
"append" is used for single values; "string-concatenation" (and list-concatenation) are used for sequences. `UTF16EncodeCodePoint` returns a list, because some code points require multiple UTF-16 code units.

Reply:
hm, ok, i guess that makes sense - just seems strange to have two different forms of the same conceptual operation.